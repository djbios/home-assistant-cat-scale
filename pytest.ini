[pytest]
asyncio_mode = auto
addopts = --cov=custom_components.cat_scale --cov-report=term-missing --cov-report=xml

[coverage:run]
branch = True