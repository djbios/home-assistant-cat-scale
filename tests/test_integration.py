import pytest
from homeassistant.helpers import entity_registry as er
from homeassistant.helpers import device_registry as dr


@pytest.mark.asyncio
async def test_entities_registered_and_available(init_integration, hass, snapshot):
    """Test that sensor entities are registered, available, and match snapshot."""
    entity_registry = er.async_get(hass)
    device_registry = dr.async_get(hass)
    entries = er.async_entries_for_config_entry(entity_registry, init_integration.entry_id)
    assert entries  # Ensure entities are registered

    # Check state and attributes for all entities
    for entry in entries:
        state = hass.states.get(entry.entity_id)
        assert state is not None
        assert state.state != "unavailable"
        assert state == snapshot

    # Check device registry
    for entry in entries:
        device = device_registry.async_get(entry.device_id)
        assert device is not None
