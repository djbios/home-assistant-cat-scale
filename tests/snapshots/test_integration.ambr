# serializer version: 1
# name: test_entities_registered_and_available
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'weight',
      'friendly_name': 'Cat weight Cat weight',
      'icon': 'mdi:cat',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': <UnitOfMass.GRAMS: 'g'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.cat_weight_cat_weight',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_entities_registered_and_available.1
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'weight',
      'friendly_name': 'Cat weight Baseline',
      'icon': 'mdi:scale-balance',
      'unit_of_measurement': <UnitOfMass.GRAMS: 'g'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.cat_weight_baseline',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '0',
  })
# ---
# name: test_entities_registered_and_available.2
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'enum',
      'friendly_name': 'Cat weight Detection state',
      'icon': 'mdi:radar',
      'options': list([
        'idle',
        'waiting_for_confirmation',
        'cat_present',
        'after_cat',
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'sensor.cat_weight_detection_state',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'idle',
  })
# ---
# name: test_entities_registered_and_available.3
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'weight',
      'friendly_name': 'Cat weight Waste weight',
      'icon': 'mdi:delete-variant',
      'unit_of_measurement': <UnitOfMass.GRAMS: 'g'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.cat_weight_waste_weight',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '0.0',
  })
# ---
